### PR TITLE
Feature - Swift4.1

### PR DESCRIPTION
**Pull request by ppeelen**
This PR will add support for Swift 4.1 and XCode 9.3

 • Swift 4.1
 • XCode 9.3 Enable:
 -> Implicit retain of self within blocks
 -> Overriding deprecated obj-c methods
 -> Whole module optimization
 • Remove weak protocol properties
 • Changes `flapMap` to `compactMap`

### CODE REVIEW

**Comment by ppeelen** [comment]
My apologies. This was not meant to be send as a PR to /linkedin/LayoutKit just yet.

File: Sources/Views/ReloadableViewUpdateManager.swift [hunk 1]
```diff
@@ -18,12 +18,12 @@ protocol ReloadableViewUpdateManagerDelegate: class {
...
 
     /// The delegate that this manager performs updates on.
-    weak var delegate: ReloadableViewUpdateManagerDelegate? { get }
+    var delegate: ReloadableViewUpdateManagerDelegate? { get }
 
     /// The operation that this update manager is associated with.
     /// The update manager will stop updating the delegate
     /// if the operation is cancelled or dellocated.
-    weak var operation: Operation? { get }
+    var operation: Operation? { get }
```
Comment (staguer) [inline Sources/Views/ReloadableViewUpdateManager.swift]: Why does this need to change?

Comment (ppeelen) [reply Sources/Views/ReloadableViewUpdateManager.swift]: Using `weak` in a protocol declaration will invoke the warning
> 'weak' should not be applied to a property declaration in a protocol and will be disallowed in future versions.

This is the result of proposal https://github.com/apple/swift-evolution/blob/master/proposals/0186-remove-ownership-keyword-support-in-protocols.md

**Comment by staguer** [comment]
@ppeelen Thank you for making these changes. Do you know what's causing the travis CI build to fail?

**Pull request closed without merging**